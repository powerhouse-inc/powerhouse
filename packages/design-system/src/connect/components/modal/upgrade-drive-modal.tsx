import { Modal } from "@/powerhouse";
import React, { ComponentPropsWithoutRef } from "react";
import { twMerge } from "tailwind-merge";

const buttonStyles =
  "min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all";

export type ConnectUpgradeDriveModalProps = ComponentPropsWithoutRef<
  typeof Modal
> & {
  readonly header: React.ReactNode;
  readonly body: React.ReactNode;
  readonly onContinue: () => void;
  readonly cancelLabel: string;
  readonly continueLabel: string;
};

export function ConnectUpgradeDriveModal(props: ConnectUpgradeDriveModalProps) {
  const {
    body,
    header,
    onOpenChange,
    onContinue,
    cancelLabel,
    continueLabel,
    overlayProps,
    contentProps,
    ...restProps
  } = props;

  return (
    <Modal
      contentProps={{
        ...contentProps,
        className: twMerge("rounded-3xl", contentProps?.className),
      }}
      onOpenChange={onOpenChange}
      overlayProps={{
        ...overlayProps,
        className: overlayProps?.className,
      }}
      {...restProps}
    >
      <div className="w-[400px] p-6 text-slate-300">
        <div className="border-b border-slate-50 pb-2 text-2xl font-bold text-gray-800">
          {header}
        </div>
        <div className="my-6 rounded-md bg-slate-50 p-4 text-center text-slate-200">
          {body}
        </div>
        <div className="mt-8 flex justify-between gap-3">
          <button
            className={twMerge(
              buttonStyles,
              "flex-1 bg-slate-50 text-slate-800",
            )}
            onClick={() => onOpenChange?.(false)}
          >
            {cancelLabel}
          </button>
          <button
            className={twMerge(buttonStyles, "flex-1 bg-gray-800 text-gray-50")}
            onClick={onContinue}
          >
            {continueLabel}
          </button>
        </div>
      </div>
    </Modal>
  );
}
