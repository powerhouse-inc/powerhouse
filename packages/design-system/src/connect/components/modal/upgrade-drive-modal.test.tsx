import { fireEvent, render, screen } from "@testing-library/react";
import { ConnectUpgradeDriveModal } from "./upgrade-drive-modal";

describe("UpgradeDriveModal Component", () => {
  it("should match snapshot", () => {
    render(
      <ConnectUpgradeDriveModal
        body="You are moving files from a private to a shared drive. These files will become accessible to others in the shared drive.Do you want to proceed?"
        cancelLabel="Cancel"
        continueLabel="Continue"
        data-testid="upgrade-drive-modal"
        header="Upgrade to cloud drive"
        onContinue={() => {}}
        open
      />,
    );

    const modalComponent = screen.getByTestId("upgrade-drive-modal");

    expect(modalComponent).toMatchSnapshot();
  });

  it("should render modal content", () => {
    const headerText = "Upgrade to cloud drive";
    const bodyText =
      "ou are moving files from a private to a shared drive. These files will become accessible to others in the shared drive.Do you want to proceed?";
    const cancelLabel = "Cancel";
    const continueLabel = "Continue";

    render(
      <ConnectUpgradeDriveModal
        body={bodyText}
        cancelLabel={cancelLabel}
        continueLabel={continueLabel}
        data-testid="upgrade-drive-modal"
        header={headerText}
        onContinue={() => {}}
        open
      />,
    );

    const header = screen.getByText(headerText);
    const body = screen.getByText(bodyText);
    const cancelButton = screen.getByText(cancelLabel);
    const deleteButton = screen.getByText(continueLabel);

    expect(header).toBeInTheDocument();
    expect(body).toBeInTheDocument();
    expect(cancelButton).toBeInTheDocument();
    expect(deleteButton).toBeInTheDocument();
  });

  it("should trigger onClose callback when cancel is clicked", () => {
    const onClose = vi.fn();
    const cancelLabel = "Cancel";

    render(
      <ConnectUpgradeDriveModal
        body="You are moving files from a private to a shared drive. These files will become accessible to others in the shared drive.Do you want to proceed?"
        cancelLabel={cancelLabel}
        continueLabel="Continue"
        data-testid="upgrade-drive-modal"
        header="Upgrade to cloud drive"
        onContinue={() => {}}
        onOpenChange={onClose}
        open
      />,
    );

    fireEvent.click(screen.getByText(cancelLabel));

    expect(onClose).toHaveBeenCalled();
  });

  it("should trigger onDelete callback when CTA is clicked", () => {
    const onDelete = vi.fn();
    const continueLabel = "Continue";

    render(
      <ConnectUpgradeDriveModal
        body="You are moving files from a private to a shared drive. These files will become accessible to others in the shared drive.Do you want to proceed?"
        cancelLabel="Cancel"
        continueLabel={continueLabel}
        data-testid="upgrade-drive-modal"
        header="Upgrade to cloud drive"
        onContinue={onDelete}
        open
      />,
    );

    fireEvent.click(screen.getByText(continueLabel));

    expect(onDelete).toHaveBeenCalled();
  });
});
