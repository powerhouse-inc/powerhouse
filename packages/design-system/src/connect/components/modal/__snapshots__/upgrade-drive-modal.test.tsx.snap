// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UpgradeDriveModal Component > should match snapshot 1`] = `
<div
  aria-describedby="radix-:r2:"
  aria-labelledby="radix-:r1:"
  class="bg-white data-[state=closed]:animate-zoom-out data-[state=open]:animate-zoom-in rounded-3xl"
  data-state="open"
  data-testid="upgrade-drive-modal"
  id="radix-:r0:"
  role="dialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <span
    style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
  >
    <h2
      id="radix-:r1:"
    >
      Modal
    </h2>
    <p
      id="radix-:r2:"
    >
      Modal
    </p>
  </span>
  <div
    class="w-[400px] p-6 text-slate-300"
  >
    <div
      class="border-b border-slate-50 pb-2 text-2xl font-bold text-gray-800"
    >
      Upgrade to cloud drive
    </div>
    <div
      class="my-6 rounded-md bg-slate-50 p-4 text-center text-slate-200"
    >
      You are moving files from a private to a shared drive. These files will become accessible to others in the shared drive.Do you want to proceed?
    </div>
    <div
      class="mt-8 flex justify-between gap-3"
    >
      <button
        class="min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all flex-1 bg-slate-50 text-slate-800"
      >
        Cancel
      </button>
      <button
        class="min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all flex-1 bg-gray-800 text-gray-50"
      >
        Continue
      </button>
    </div>
  </div>
</div>
`;
